pkg/controllers/user/rbac/namespace_handler.go

```diff
@@ -87,6 +88,14 @@ func (n *nsLifecycle) Remove(obj *v1.Namespace) (runtime.Object, error) {
 }
 
 func (n *nsLifecycle) syncNS(obj *v1.Namespace) (bool, error) {
+	if err := n.cleanRBFromNamespace(obj); err != nil {
+		return false, err
+	}
+
+	if err := n.ensureInjectedAppRBAddToNamespace(obj); err != nil {
```

Comment:
I'm not sure if putting the ensureInjectedAppRBAddToNamespace to the syncNS is a good idea. First, this method is generic, and shouldn't have any app specific logic. Second, it wouldn't handle the case when app was added to an existing ns. It would only take care of the ns events, when ns gets created or updated. 

If we are to add app specific rbac, I'd add it separately. And trigger it on system apps events only.